### PR TITLE
Add instructions for building a custom collector

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
This PR closes [Issue 4126](https://github.com/elastic/observability-docs/issues/4126) and adds steps for building a custom collector.

Things to figure out:

- [x] Can we use the same config for MacOS and Linux
- [x] Complete K8s steps

### CODE REVIEW

**Review by mlunadia — CHANGES_REQUESTED** [review]
We should not provide users instructions to build a collector per platform but instructions to build a collector that has all the EDOT components and then provide them with options to configure it for different use cases.

File: docs/configure-upstream-collector.md [hunk 1]
```diff
@@ -1,10 +1,59 @@
...
 Refer to the [guided onboarding](guided-onboarding.md) docs or the [manual configuration](manual-configuration.md) docs for more on configuring the Elastic Distribution of the OpenTelemetry Collector.
 
 ## Upstream collector configuration examples
-Refer to the OpenTelemetry documentation on [building a custom collector](https://opentelemetry.io/docs/collector/custom-collector/) for more on creating an upstream collector.
-Use the Elastic [example configurations](https://github.com/elastic/elastic-agent/tree/main/internal/pkg/otel/samples) as a reference when configuring your upstream collector.
\ No newline at end of file
+
+Use the Elastic [example configurations](https://github.com/elastic/elastic-agent/tree/main/internal/pkg/otel/samples) as a reference when configuring your upstream collector.
+
+## Build a custom collector for MacOS or Linux
```
[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply docs/configure-upstream-collector.md:12]: I structured it in this way according to @ChrsMark comment [here](https://github.com/elastic/observability-docs/issues/4126#issuecomment-2275333268). Perhaps @ChrsMark and @ishleenk17 can provide feedback on this PR on how to provide this info with all of the options?

[the remark this answers is not in the document]
Comment (mlunadia) [reply docs/configure-upstream-collector.md:12]: @AlexanderWert can you confirm this one? AFAIK is a single binary with different configs, not a binary per platform.

[the remark this answers is not in the document]
Comment (AlexanderWert) [reply docs/configure-upstream-collector.md:12]: The builder itself has different binaries for different platforms, see here: https://opentelemetry.io/docs/collector/custom-collector/#step-1---install-the-builder

But the rest of the building instructions should be identical.

[the remark this answers is not in the document]
Comment (mlunadia) [reply docs/configure-upstream-collector.md:12]: ah makes sense.
@mdbirnstiehl so what you have now makes sense but I feel is off to have a different flow for each platform ideally we want to have a single flow with different options as per the link shared above.

[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply docs/configure-upstream-collector.md:12]: @mlunadia Got it, I'll update to one flow.

[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply docs/configure-upstream-collector.md]: @mlunadia @AlexanderWert I've update the doc to follow a single flow. I'm wondering if the info on the OpenTelemetry Collector Contrib distribution should still be on this page or if it makes sense to move that to its own page. Do we need to add specific steps on updating the collector contrib?

**Review by mlunadia — APPROVED** [review]
LGTM

**Pull request merged**